ltm/agent_think.py

```diff
@@ -0,0 +1,43 @@
+from dataclasses import dataclass
+from typing import Callable
+from ltm.agent import LTMAgent
+from utils.ui import colour_print
+from utils.llm import LLMContext, make_user_message, make_assistant_message
+
+
+@dataclass
+class LTMReflexionAgent(LTMAgent):
+
+    def _truncated_completion(self, context: LLMContext, max_messages: int = None, **kwargs) -> str:
+        max_messages = max_messages or len(context) + 1
+        while len(context) + 1 > max_messages or self.count_tokens(messages=context) > self.max_input_tokens:
```

Comment:
Are any of the methods in this subclass ever used?

Reply:
No. I just let them there for when we have the new evaluation tool. But now that you mention it, it seems dumb. I'll just save the file in my stash and remove it from here.